modules/ossm-cert-manager-installing-istio-resource.adoc

```diff
@@ -0,0 +1,150 @@
+// Module included in the following assemblies:
+//
+// * service-mesh-docs-main/install/ossm-cert-manager-assembly.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="installing-istio-resource_{context}"]
+= Installing your Istio resource
+
+//TP1 content influx. Title, etc may change.
+//Content is very similar to 2.x content
+//all kinds of formatting things to fix. want to see if a build will generate to have a look, and see how it fits structurally with the IA.
+
+
+After you have installed `istio-csr` by following the procedure for either `InPlace` or `RevisionBased`, you can install the `Istio` resource.
+
+You need to disable Istio's built in CA server and tell istiod to use the `istio-csr` CA server. The `istio-csr` CA server issues certificates for both istiod and user workloads.
+
+.Procedure
+
+. Create the `Istio` object as in the following example:
++
+.Example `istio.yaml` object
+[source, yaml]
+----
+apiVersion: sailoperator.io/v1alpha1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: v1.23.0
+  namespace: istio-system
+  values:
+    global:
+      caAddress: cert-manager-istio-csr.cert-manager.svc:443
+    pilot:
+      env:
+        ENABLE_CA_SERVER: "false"
+      volumeMounts:
+        - mountPath: /tmp/var/run/secrets/istiod/tls
+          name: istio-csr-dns-cert
+          readOnly: true
+----
++
+[NOTE]
+====
+If you followed the instructions for "istio-csr installation: RevisionBased update strategy", then you need to add the following to your `Istio` object yaml:
+
+[source, yaml]
+----
+kind: Istio
+metadata:
+  name: default
+spec:
+  updateStrategy:
+    type: RevisionBased
+----
+====
+
+. Create the `Istio` resource using the following command:
++
+[source, terminal]
+----
+$ oc apply -f istio.yaml
+----
+
+. Wait for `Istio` to become ready:
++
+[source, terminal]
+----
+$ oc wait --for=condition=Ready istios/default -n istio-system
+----
+
+.Verification
+
+Use the sample `httpbin` service and `sleep` app to check communication between the workloads is possible and to check the workload certificate of the proxy to verify that the cert-manager tool is installed correctly.
+
+. Create the `sample` namespace:
++
+[source, terminal]
+----
+$ oc new-project sample
+----
+
+. Find your active `IstioRevision`:
++
+[source, terminal]
+----
+$ oc get istiorevisions
+----
+
+. Add the injection label for your active revision to the `sample` namespace:
++
+[source, terminal]
+----
+$ oc label namespace sample istio.io/rev=<your-active-revision-name> --overwrite=true
+----
+
+. Deploy the sample `httpbin` app:
++
+[source, terminal]
+----
+$ oc apply -n sample -f https://raw.githubusercontent.com/istio/istio/refs/heads/master/samples/httpbin/httpbin.yaml
```

Comment:
A lot of this came from https://docs.openshift.com/container-platform/4.17/service_mesh/v2x/ossm-security.html#ossm-cert-manager-installation_ossm-security but since Service Mesh switched to `/istio/` it says `/istio/` instead of `/maistra/`

Reply:
How is this file reviewed? The concern with applying yaml files out of GitHub is that they are generally uncontrolled. The rule is to link to them out of the openshift org because we know that all changes go through QE.

Reply:
@nrfox , there are concerns about sending users to raw.githubusercontent files

Reply:
> How is this file reviewed? The concern with applying yaml files out of GitHub is that they are generally uncontrolled. The rule is to link to them out of the openshift org because we know that all changes go through QE.

[This file](https://github.com/istio/istio/tree/master/samples/httpbin) is reviewed by the Istio org which does include some RHers but also some non RH folks. We can use the [openshift-service-mesh fork](https://github.com/openshift-service-mesh/istio) for these samples instead.

Reply:
```suggestion
$ oc apply -n sample -f https://raw.githubusercontent.com/openshift-service-mesh/istio/refs/heads/master/samples/httpbin/httpbin.yaml
```